Remove more than half the stack usage of CFG traversal

dberlin opened the pull request:
A lot of recursive stack space is used walking the statements in order, which is not necessary.

dberlin commented:
Sorry, missed staging a hunk on the reverse one, one minute

s3rvac commented on src/llvmir2hll/graphs/cfg/cfg_traversal.cpp:
```diff
@@ -168,7 +168,7 @@ bool CFGTraversal::getCurrRetVal() const {
 */
```
The descriptions of the two modified functions are
```
* @brief A recursive implementation of performTraversal().
```
```
* @brief A recursive implementation of performReverseTraversal().
```
I guess we should remove the word _recursive_ from them as they are no longer recursive.

dberlin replied:
Changed to non-recursive

dberlin commented:
Sorry, i'll fix.
I have a follow on patch that removes all the recursion (the version here still takes about 30+ gig of stack on some of my test cases), and i updated the comments in there :)

dberlin commented:
I updated the comment. Note that they are still recursive, as traverseNodeSuccessors will recurse back into calling us.

The traversal of CFG nodes as implemented is quite weird in actual practice. I rewrote it in a followup, i'll include more details there.

s3rvac commented:
That AppVeyor build failure is due to a timeout and has nothing to do with this PR. We will have to figure what to do with it.
